### ISSUE TITLE
Use fully qualified path for Results

### ISSUE THREAD

**Issue by silvasch** [ryo33/enum_downcast#2]
Hey!

This is an awesome crate.

However, I had one issue, specifically with this piece of code:

```rust
use enum_downcast::EnumDowncast;

struct CustomError;
type Result<T> = std::result::Result<T, CustomError>;

#[derive(EnumDowncast)]
enum EnumWithDowncast {
    Player(i32),
    Enemy(String),
}
```

This causes compilation to fail, because the `EnumDowncast` derive macro contains `Result` (e. g. [here](https://github.com/ryo33/enum_downcast/blob/756935d55a90d6e496a82fec18ad31ca80cdfd5d/crates/enum_downcast_derive/src/variant.rs#L47)) and gets overridden by the custom `Result` implementation.

This would probably be fixed by using the fully qualified path to `Result` in the `enum_downcast_derive` crate (Replace every `Result<...>` with `std::result::Result<...>`).

If you want, I can open a PR for this.

**Comment by ryo33** [comment]
Thanks for your suggestion!

Yes, it should use a qualified path. And, sending a PR is appreciated! since this crate would love to support `no_std`, the fully qualified paths should be `core::result::Result` instead of `std::` one.

**Comment by silvasch** [comment]
Great! I'll submit a PR sometime today.

**Comment by silvasch** [comment]
Implemented in 0.2.0
